### PR TITLE
ICMSLST- 1847 Restructure icms-hmrc codebase.

### PR DESCRIPTION
**Pull request by MattHolmes123**

### CODE REVIEW

File: mail/management/commands/resend_email.py [hunk 1]
```diff
@@ -90,6 +92,42 @@ def handle(self, *args, **options):
...
+        # This is the case where we sent a licence_data email earlier which hasn't reached HMRC
+        # and so we are resending it
+        sender = settings.EMAIL_USER
+        receiver = settings.OUTGOING_EMAIL_USER
+        attachment = [mail.sent_filename, mail.sent_data]
+    else:
+        raise Exception(f"Extract type not supported: {mail.extract_type}")
+
+    logger.info(
+        "Preparing request Mail dto of extract type %s, sender %s, receiver %s with filename %s",
```
Comment (mikeylockwood) [inline mail/management/commands/resend_email.py:117]: Is dto a typo?

Comment (mikeylockwood) [reply mail/management/commands/resend_email.py:117]: No it's in the function name

File: mail/tasks.py [hunk 2]
```diff
@@ -59,6 +70,286 @@ def send_licence_data_to_hmrc_shared() -> Optional[bool]:
...
+
+    logger.info("Updating Mail %s status from %s => %s", mail.id, previous_status, mail.status)
+
+    mail.save()
+
+
+def _get_licence_reply_data(processor: LicenceReplyProcessor) -> Dict[str, Any]:
+    # Load all LicencePayload records linked to the current LicenceData record
+    ld: LicenceData = LicenceData.objects.get(hmrc_run_number=processor.file_header.run_num)
+    current_licences = ld.licence_payloads.all().values_list("lite_id", "reference", named=True)
```
Comment (mikeylockwood) [inline mail/tasks.py:221]: Classic `.all()`

Comment (MattHolmes123) [reply mail/tasks.py:221]: I'm getting it on a tshirt

**Review by mikeylockwood — APPROVED** [review]

**Pull request merged**